exercises/practice/all-your-base/.approaches/introduction.md

```diff
@@ -0,0 +1,68 @@
+# Introduction
+
+The process of the _all your base_ exercise can be divided into  up to four parts:
+- conversion from input digit vector into a base 10 number
+- conversion from intermediate number into target base vector
+- possible reversing of the resulting vector
+- argument error checking
+
+## General structure
+
+The key to solving all your base is to handle one digit at a time.
+You will use the [remainder operator][remainder-operator] repeatedly in tandem with the [division operator][division-operator] or make use of the [`pow`][pow-function].
+
+Algorithms for converting to an intermediate base 10 `int` can be grouped by possible performance.
+Some approaches can be parallelized, others might use fewer or less expensive operations.
+
+As the structure of the returning vector is build up in order, while modifying the same integer, there is no potential for a parallel process.
+
+Reversing standard containers like vectors works the same way as discussed in the [reverse-string][reverse-string] exercise.
+Depending on the algorithm to build the outgoing vector, the reversion might be unnecessary.
+
+~~~~exercism/note
+You might have heard Donald Knuth's famous words ["Premature Optimization Is the Root of All Evil"][pre-opt-blog].
+With all the talk about best performance, please take care of the readability and therefore maintainability of your code.
+~~~~
+
+## Approach: Converting to base 10
+
+```cpp
+unsigned int intermediate = 0;
+for (unsigned int d : input_digits) {
+    intermediate = intermediate * input_base + d;
+}
+```
+
+For more information, check the [converting to base 10 approach][to-base-ten].
+
+## Approach: Converting from base 10
+
+```cpp
+while (intermediate != 0) {
+    unsigned int new_digit = intermediate % output_base;
+    intermediate = (unsigned int)(intermediate / output_base);
```

Comment:
Is this cast really necessary, aren't `intermediate` and `output_base` both `unsigned int`?
If it is necessary, please use a named cast instead of a C-style cast.